audb/core/publish.py

```diff
@@ -644,32 +646,25 @@ def publish(
 
     # dependencies do not match version
     if previous_version is not None and len(deps) > 0:
-        with tempfile.TemporaryDirectory() as tmp_dir:
-            previous_deps_path = os.path.join(
-                tmp_dir,
-                define.DEPENDENCIES_FILE,
-            )
-            previous_deps = dependencies(
-                db.name,
-                version=previous_version,
-                cache_root=cache_root,
-                verbose=verbose,
+        previous_deps = dependencies(
+            db.name,
+            version=previous_version,
+            cache_root=cache_root,
+            verbose=verbose,
+        )
+        if not deps().equals(previous_deps()):
```

Comment:
You mention the observation that the md5sum equality does not hold any longer when applied to files persisted. Do we know anything about the origins of that behavior: [This link](https://stackoverflow.com/questions/74723209/parquet-file-re-write-has-slightly-larger-size-in-both-pandas-pyarrow) and following to [there](https://groups.google.com/g/snappy-compression/c/W8v_ydnEPuc\
?pli=1) suggests that pyarrow's default compression algorithm `snappy` should be consistent across runs, but not across library versions and possible architectures. Therefore it looks like a great move to compare the contents of the tables directly without relying on their signature on disk.

In this fashion this becomes independent of the exact compression used (gzip is possibly more stable).

I have a question about the implementation: would it not be idiomatic to implement  `def __eq__(self, other):` in `Dependencies` rather than calling `pd.DataFrame.equals` directly?

Reply:
I don't know why the dependency table files might change their MD5 sums when stored as parquet files.
For the dependency table itself, I think it is anyway better to compare the content of the table instead the stored file.
For that reason I implemented `__eq__()` in https://github.com/audeering/audb/pull/381 as you suggested.

But I agree, that the changing MD5 sum is a bigger issue when we start storing tables as parquet files as well, as all of the automatic versioning of `audb` relies on MD5 sums. Hence, they should only change if the content of the table changes. In that case, we would need to find a workaround or cannot use parquet files for storing tables.

Reply:
`__eq__` has been implemented in  https://github.com/audeering/audb/pull/381 that has been reviewed. 
But what should one do with the md5sum problem? Are there plans to store tables into parquet files at all? If not then I would resolve then conversation here. If yes then would it be interesting to identify a good compression that leaves md5sum constant across runs and library versions, or discuss possible workarounds?

Reply:
As the MD5SUM problem is only relevant for tables and not for the dependency table, I added a discussion to https://github.com/audeering/audformat/issues/376#issuecomment-2037395916.
I think, we should continue the discussion there as it has no influence on the dependency tables of `audb` as we found another way to compare the tables here.